contracts/base/BaseLiquidityManagement.sol

```diff
@@ -137,9 +188,24 @@ contract BaseLiquidityManagement is IBaseLiquidityManagement, SafeCallback {
         uint256 liquidityToAdd,
         bytes memory hookData,
         bool claims
-    ) internal returns (BalanceDelta delta) {
-        delta = _increaseLiquidity(owner, range, liquidityToAdd, hookData);
-        zeroOut(delta, range.key.currency0, range.key.currency1, owner, claims);
+    ) internal returns (BalanceDelta callerDelta) {
+        callerDelta = _increaseLiquidity(owner, range, liquidityToAdd, hookData);
+        _closeCallerDeltas(callerDelta, range.poolKey.currency0, range.poolKey.currency1, owner, claims);
+        _closeAllDeltas(range.poolKey.currency0, range.poolKey.currency1);
+    }
+
+    // When chaining many actions, this should be called at the very end to close out any open deltas owed to or by this contract for other users on the same range.
+    // This is safe because any amounts the caller should not pay or take have already been accounted for in closeCallerDeltas.
+    function _closeAllDeltas(Currency currency0, Currency currency1) internal {
+        int128 delta0 = int128(manager.currencyDelta(address(this), currency0));
+        int128 delta1 = int128(manager.currencyDelta(address(this), currency1));
```

Comment:
you can use CurrencyDeltas, which uses sparse extload to load both deltas in a single external call 😎

Reply:
nice!

Reply:
can we move that function to TransientStateLibrary.sol ?